pan/strings.pan

```diff
@@ -0,0 +1,22 @@
+declaration template pan/strings;
+
+
+@documentation{
+Convenience type specifying a string that must contain at least one character (anything other than an empty string).
+Intended for use in building more complex types.
+}
+type string_non_empty = string with match(SELF, '^.+$');
```

Comment:
`'.'` is enough, you can also test the lenght of the string; that probably even faster